core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -92,14 +94,17 @@ class CityConstructions {
      * @return Maintenance cost of all built buildings
      */
     fun getMaintenanceCosts(): Int {
-        var maintenanceCost = getBuiltBuildings().sumBy { it.maintenance }
```

Comment:
Clear approach, but my nose tingles... getListOfFreeBuildings(cityInfo.id) returns a Set<String> - MutableSet interface, acutally the LinkedHashSet implementation if I'm not mistaken. Lookups should be _fast_. So - cache the set reference, loop buildings, sum if !in set should be a few percent faster.

Reply:
This could be changed, but I do not feel like the speedup would be massive. Might look into it nonetheless. For now, the new version over at #4106 has cached the map of maintenance-free buildings for each cities, which should also improve performance.

Reply:
I've changed it after all